### ISSUE TITLE
transformers DataCollatorWithPadding should have an option to specify the attributes which require padding.

### ISSUE THREAD

**Issue by varadhbhatnagar** [huggingface/transformers#26758]
### Feature request

Currently there is no way to specify the attributes which require padding in transformers [DataCollatorWithPadding](https://github.com/huggingface/transformers/blob/v4.34.0/src/transformers/data/data_collator.py#L215).

### Motivation

Two instances from our data may be of this format:
```
{
"input_ids": [1,2,3,4,5],
"attention_mask": [0,1,1,1,1],
"label": "apple"
}

{
"input_ids": [1,2,3,4,5,6,7,8],
"attention_mask": [0,1,1,1,1,1,1,1],
"label": "mango"
}

```
User wants to implement [dynamic padding](https://huggingface.co/learn/nlp-course/chapter3/2?fw=pt#dynamic-padding) over these two data points, but currently there is no way to specify that `label` attribute should not be passed through [`self.tokenizer.pad()`]()https://github.com/huggingface/transformers/blob/b71f20a7c9f3716d30f6738501559acf863e2c5c/src/transformers/data/data_collator.py#L249

### Your contribution

I can submit a PR for this.
